### PR TITLE
Update CODEOWNERS to add docs team as reviewers for docs changes

### PR DESCRIPTION
**Pull request by theletterf**
As discussed in Slack, this PR aims at standardizing docs team as reviewers for all changes concerning documentation in our repos.

### CODE REVIEW

**Review by MrAlias — APPROVED** [review]

File: specification/templates/.github/CODEOWNERS [hunk 1]
```diff
@@ -4,3 +4,14 @@
...
+#####################################################
+#
+# Docs reviewers
+#
+#####################################################
+
+*.md @signalfx/docs
+*.rst @signalfx/docs
+docs/ @signalfx/docs
+README @signalfx/docs @signalfx/gdi-specification-approvers @signalfx/gdi-specification-maintainers
```
Comment (pellared) [inline specification/templates/.github/CODEOWNERS:17]: I think most often it is `README.md`.

```suggestion
README* @signalfx/docs @signalfx/gdi-specification-approvers @signalfx/gdi-specification-maintainers
```

**Review by flands — APPROVED** [review]

File: specification/templates/.github/CODEOWNERS [hunk 2]
```diff
@@ -4,3 +4,14 @@
...
+#####################################################
+#
+# Docs reviewers
+#
+#####################################################
+
+*.md @signalfx/docs
+*.rst @signalfx/docs
+docs/ @signalfx/docs
+README* @signalfx/docs @signalfx/gdi-specification-approvers @signalfx/gdi-specification-maintainers
```
Comment (mateuszrzeszutek) [inline specification/templates/.github/CODEOWNERS:17]: That will create PRs that do not have `@signalfx/gdi-TEAM-maintainers @signalfx/gdi-TEAM-approvers` as reviewers. Is that really correct? I think that it's great for the docs team to review those files, but the repo maintainers should also be added - to ensure that the docs are correct and match the actual code in the repo.

Comment (MrAlias) [reply specification/templates/.github/CODEOWNERS:17]: @mateuszrzeszutek can you open a PR to fix this?

Comment (mateuszrzeszutek) [reply specification/templates/.github/CODEOWNERS:17]: https://github.com/signalfx/gdi-specification/pull/117

**Pull request merged**